Sage/carbon
# fix(multi-action-button, split-button): fix aria role

The ARIA role of the child buttons was inadvertently set to `menu-item` instead of the correct `menuitem`, thus causing accessibility failings

fix #6383

### Proposed behaviour



No Axe errors should occur when testing with the child buttons open

### Current behaviour

Multiple "critical" Axe violations are found: 
![image](https://github.com/Sage/carbon/assets/25186376/18fe044b-b11e-40bd-a215-d52dca90d543)




### Checklist




#### QA


### Additional context



### Testing instructions

Run Axe devtools with the buttons open (easiest if they are opened using the keyboard) and verify no errors. (Note, you may see some about "all page content should be contained by landmarks" and other landmark-related errors. These only appear if you have "best practices" turned on in Axe devtools and are only "moderate" severity. They need further discussion in the team as to whether they need fixing - they appear regardless of the changes in this PR so are not a reason to reject it.)

There should definitely be no "critical" errors and none referring to ARIA roles.

Test on both MultiActionButton and SplitButton.
